src/common/asb/Asb.c

```diff
@@ -1678,7 +1678,7 @@ static char* AuditEnsureSourceRoutedPacketsIsDisabled(void* log)
 static char* AuditEnsureAcceptingSourceRoutedPacketsIsDisabled(void* log)
 {
     char* reason = 0;
-    RETURN_REASON_IF_NOT_ZERO(CheckLineFoundNotCommentedOut("/proc/sys/net/ipv4/conf/all/accept_source_route", '#', "0", &reason, log));
+    RETURN_REASON_IF_NOT_ZERO(CheckLineFoundNotCommentedOut("/proc/sys/net/ipv4/conf/default/accept_source_route", '#', "0", &reason, log));
```

Comment:
We need both 'all' and 'default' for this rule. This is the code in main:

```
static char* AuditEnsureAcceptingSourceRoutedPacketsIsDisabled(void* log)
{
    char* reason = 0;
    RETURN_REASON_IF_NOT_ZERO(CheckLineFoundNotCommentedOut("/proc/sys/net/ipv4/conf/all/accept_source_route", '#', "0", &reason, log));
    CheckLineFoundNotCommentedOut("/proc/sys/net/ipv6/conf/default/accept_source_route", '#', "0", &reason, log);
    return reason;
}
````

It matches the ASB recipe and the remediation function in same file:

```
static int RemediateEnsureAcceptingSourceRoutedPacketsIsDisabled(char* value, void* log)
{
    UNUSED(value);
    return (SavePayloadToFile("/proc/sys/net/ipv4/conf/all/accept_source_route", "0", 1, log) &&
        SavePayloadToFile("/proc/sys/net/ipv6/conf/default/accept_source_route", "0", 1, log)) ? 0 : ENOENT;
}
```

Please let me know if I did not understood correctly what you are trying to do.

Reply:
That is the bug - `net.ipv4.conf.all.accept_source_route` and `net.ipv6.conf.all.accept_source_route` are already checked by
`AuditEnsureSourceRoutedPacketsIsDisabled`/`RemediateEnsureSourceRoutedPacketsIsDisabled` (https://github.com/Azure/azure-osconfig/blob/main/src/common/asb/Asb.c#L1670-L1676, 
https://github.com/Azure/azure-osconfig/blob/main/src/common/asb/Asb.c#L3263-L3268).

`AuditEnsureAcceptingSourceRoutedPacketsIsDisabled`/`RemediateEnsureAcceptingSourceRoutedPacketsIsDisabled` should be checking `net.ipv4.conf.default.accept_source_route` and `net.ipv6.conf.default.accept_source_route`. These are the two functions I am correcting here.

Reply:
Understood. As per these two entries at https://learn.microsoft.com/en-us/azure/governance/policy/samples/guest-configuration-baseline-linux:

<html>
<body>
<!--StartFragment-->
The default setting for accepting source routed packets should be disabled for network interfaces. (net.ipv4.conf.default.accept_source_route = 0)(42.1) | Description: An attacker could redirect traffic for malicious purposes. | Run sysctl -w key=value and set to a compliant value.
-- | -- | --
The default setting for accepting source routed packets should be disabled for network interfaces. (net.ipv6.conf.default.accept_source_route = 0)(42.2) | Description: An attacker could redirect traffic for malicious purposes. | Run sysctl -w key=value and set to a compliant value.

<!--EndFragment-->
</body>
</html>